fluent-kafka-streams-tests/src/main/java/com/bakdata/fluent_kafka_streams_tests/TestTopology.java

```diff
@@ -223,14 +270,41 @@ public <K, V> TestTopology<K, V> withDefaultSerde(final Serde<K> defaultKeySerde
      * @return Copy of current {@code TestTopology} with provided {@code SchemaRegistryMock}
      */
     public TestTopology<DefaultK, DefaultV> withSchemaRegistryMock(final SchemaRegistryMock schemaRegistryMock) {
-        return this.with(this.topologyFactory, this.properties, this.defaultKeySerde, this.defaultValueSerde,
+        return this.with(this.topologyFactory, this.propertiesFactory, this.defaultKeySerde, this.defaultValueSerde,
                 schemaRegistryMock);
     }
 
-    protected <K, V> TestTopology<K, V> with(final Function<? super Properties, ? extends Topology> topologyFactory,
-            final Map<?, ?> properties, final Serde<K> defaultKeySerde, final Serde<V> defaultValueSerde,
-            final SchemaRegistryMock schemaRegistry) {
-        return new TestTopology<>(topologyFactory, properties, defaultKeySerde, defaultValueSerde, schemaRegistry);
+    /**
+     * Start the {@code TestTopology} and create all required resources.
+     * <p>
+     * This method starts the {@link SchemaRegistryMock}, creates the state directory and creates a
+     * {@link TopologyTestDriver}.
+     */
+    public void start() {
+        this.schemaRegistry.start();
+        this.properties.putAll(this.propertiesFactory.apply(this.getSchemaRegistryUrl()));
```

Comment:
I think we should still set the schema registry url as a property here. Otherwise, we rely on the user to set the property like you do here: https://github.com/bakdata/fluent-kafka-streams-tests/pull/79/files#diff-d8e535869424621cdf213f7e70b66b86c8678429cf6a74f5cb280149186e525cR92
We sometimes don't set it as an application property but only for a specific serde config.

```suggestion
        this.properties.put(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, this.getSchemaRegistryUrl());
        this.properties.putAll(this.propertiesFactory.apply(this.getSchemaRegistryUrl()));
```

Reply:
It is done automatically when using the old API (i.e., providing a properties object) https://github.com/bakdata/fluent-kafka-streams-tests/pull/79/files#diff-ef53297934de3a362494075df51a9c76054f2105d902dcaa70d8b9bdbf24bf8bR156. Only for the new API the user has to take care of configuring the schema registry url using the provided URL

Reply:
But why should they take care of it when we can just set it ourselves?

Reply:
Because maybe a user wants to set it differently. This prevents conflicts if a user does not want to set it or it could be set twice